src/org/zaproxy/zap/extension/frontEndScanner/ExtensionFrontEndScanner.java

```diff
@@ -0,0 +1,358 @@
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public int getArrangeableListenerOrder() {
+        // Need to run after the HistoryReference has been saved to the database
+        return ProxyListenerLog.PROXY_LISTENER_ORDER + 42;
+    }
+
+    private String frontEndScannerCode() throws IOException {
+        Path frontEndScannerPath = Paths.get(FRONT_END_SCANNER);
+        return readFromFile(frontEndScannerPath);
+    }
+
+    private String userScriptsToInject() throws IOException {
+        Path scriptFolderPath = Paths.get(SCRIPTS_FOLDER + "client-side-passive" + '/');
+        Stream<String> scriptCodes;
+        String result = "";
+
+        try {
+            List<String> passiveFunctionNames = new ArrayList<String>();
+
+            Stream<Path> scriptFilePaths = this.extensionScript
+                .getScripts(ExtensionFrontEndScanner.SCRIPT_TYPE_CLIENT_PASSIVE)
+                .stream()
+                .filter(script -> script.isEnabled())
+                .map(script -> script.getFile())
+                .map(file -> file.toPath());
+
+            scriptCodes = scriptFilePaths
+                .map(scriptFileName -> readFromFile(scriptFileName))
+                .map(code -> code.getBytes())
+                .map(code -> new String(code))
+                // `wrapInFunction` has a side-effect: it updates the `passiveFunctionNames` List
+                .map(code -> wrapInFunction(code, passiveFunctionNames));
+
+            result = scriptCodes.reduce(result, String::concat);
+            result += "const SCRIPTS = [ " + String.join(", ", passiveFunctionNames) + "];";
+            return result;
+        } catch (UncheckedIOException e) {
+            new IOException(e);
+        }
+
+        return result;
+    }
+
+    private String readFromFile(Path file) throws UncheckedIOException {
+        try {
+            byte[] content = Files.readAllBytes(file);
+            return new String(content);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private String wrapInFunction(String javascriptCode, List<String> functionNames) {
+        String id = Long.toString(Math.abs(UUID.randomUUID().getMostSignificantBits()));
+        String functionName = "f_" + id;
+
+        functionNames.add(functionName);
+
+        return new String(
+            "function " + functionName + " (frontEndScanner) { " + javascriptCode + " };"
+        );
+    }
+
+    private ImageIcon createScriptIcon(String path) {
+        if (getView() == null) {
+            return null;
+        }
+        return new ImageIcon(
+            ExtensionFrontEndScanner.class.getResource(path));
+    }
+
+    private void addScriptsToPanel(ScriptType scriptType) {
+        try {
+            String folder = SCRIPTS_FOLDER + scriptType.getName() + '/';
+            Path scriptFolderPath = Paths.get(folder);
+            Stream<Path> scriptFilePaths = Files.list(scriptFolderPath);

```

Comment:
Eclipse is telling me that this is never closed

Reply:
Done.